scRNA-seq/01-normalizing_scRNA-seq.Rmd

```diff
@@ -92,72 +87,133 @@ plot.
 
 ```{r}
 # Let's calculate the gene means and make a density plot
-qplot(apply(seurat_obj@raw.data, 1, mean), geom = "density", xlab = "counts")
+gene_means <- apply(counts(sce_raw), 1, mean)
+
+# Plot the means
+qplot(gene_means, geom = "density", xlab = "counts") + 
+  geom_density(fill = "lightblue")
 ```
 
-As you'll notice, the unnormalized, counts data has so many zeroes. 
+The unnormalized, counts data has so many zeroes. 
 Actually for this dataset, 78.8% of the data points are zeroes, which is not 
 out of the ordinary for single-cell RNA-seq data. 
+So we can get a little bit more sensible of a graph, let's plot the same thing
+but log transformed.
+
+```{r}
+# Plot the log(means)
+qplot(log(1 + gene_means), geom = "density", xlab = "counts") + 
+  geom_density(fill = "lightblue")
+```
+
+As we should expect, our single-cell RNA-seq transcript counts consists of a few
+genes that are expressed a lot, and the rest of the genes are barely detected. 
 
 ## Normalization of count data 
 
-As we should expect, our single-cell RNA-seq transcript counts are largely from 
-a few genes that are expressed a lot, and a lot of genes that are barely 
-expressed at all.
-One way to normalize this data and make it more useful to us, is to log
-transform it.
-These are the default settings for normalization and should work in most 
-instances. 
+In whatever data we are working with, we are always looking to maximize biological 
+variance and minimize technical variance. 
+This is where normalization methods usually come into the workflow.
+For our data, many of these zeroes are likely biologically sensible: an individual
```

Comment:
To me, it makes sense to frame this as reducing technical variance such that individual cells are more comparable. We are not trying to get rid of zeros. (TIL zeros is the plural of zeros and zeroes is a verb. I have been wrong for so long.) My understanding is that the zeros do make accounting for the differences between cells more difficult (and why normalization would be different from bulk) and that is what `scran` is trying to solve.

Reply:
Yeah, the way I wrote this sounds like we are "getting rid of zero(e)s" but we aren't. Hm. Let me go back to the paper and try to come up with more accurate wording that also hopefully will still make sense to our audience. Also some dictionaries have "zeroes" as a noun. I think it is a similar thing that happened with data being plural. Enough people used it wrong that eventually it became right. The word "zeros" makes me viscerally uncomfortable, even if it's not "supposed" to be noun.